lib/rom/sql/attribute.rb

```diff
@@ -235,6 +257,11 @@ def sql_expr
         @sql_expr ||= (meta[:sql_expr] || Sequel[to_sym])
       end
 
+      # @api private
+      def type_extensions
+        @type_extensions ||= TypeExtensions[unwrap_type(type)] || EMPTY_HASH
+      end
```

Comment:
I thought about it, this is not gonna work, consider this

```ruby
schema do
  attribute :foo, Types::String.optional
end
```
The `attribute` method accepts an already unwrapped type, we have to unwrap it anyway